rfcs/0008-distributed-dns.md

```diff
@@ -0,0 +1,274 @@
+
+```
+{
+  Type: UnresolvedConflict
+  Status: False
+  Reason: FirstValidation
+  Message: "listener: a.b.com validated successfully. re-queue validation for <time>"
+}
+{
+  Type: UnresolvedConflict
+  Status: True
+  Reason: RecordsMissing|ConflictingStrategy|ConflictingValue
+  Message: "listener: a.b.com strategy : simple conflicts. Current validation atttempt is y in z minutes. Next is <time>"
+}
+{
+  Type: UnresolvedConflict
+  Status: True
+  Reason: RecordsMissing|ConflictingStrategy|ConflictingValue
+  Message: "listener: a.b.com health check changed : Current validation atttempt is y in z minutes. Next is <time>"
+}
+
+```
+
+### How do we spot misconfiguration / on-going conflicts
+
+Misconfiguration is focused around the DNSPolicy (as this is our Kuadrant API). However other forms of misconfig/conflict may also be spotted with the proposed method. 
+With this type of conflict, the controller will observe an unresolvable state. This will present itself as there having been no change to the spec of the local DNSRecord (observed by the generation sequence number matching the observedGeneration in the status) but the remote values in the zone have changed. When this is spotted we will update the status as above and increment a metric to count encountered conflicts, we may also decide to further increase the requeue time to reduce load. Issue has been reported and so back off until issue resolved.
+An example of this would be if you had one cluster with a DNSPolicy whose strategy was set to LoadBalanced and second cluster with a DNSPolicy that had a strategy of Simple (assuming this DNSPolicy was targeting a listener with the same hostname). This type of configuration would mean the structure of the DNS zone would constantly change. In a strategy conflict, the controller will be able to observe this due to the structure of the remote record set. In this case, the controller will not overwrite but instead will report the issue and requeue.
+
+
+### Configuration Updates
+There will be a new configuration option that can be applied as runtime argument (to allow us emergency configuration rescue in case of an unexpected issue) to the kuadrant-operator:
+- `clusterID`  (see below for how this is used)
+  - When absent, this is generated in code (more information below). A unique string to identify this cluster apart from all other clusters.
+
+### Changes to DNS Policy reconciler (kuadrant operator)
+
+This will primarily need to be made aware that it will need to propagate the DNS Policy health check block into any DNS Records that it creates or updates using the provider specific fields.
+
+It will also need to update the format of the records in the DNSRecord it creates. We will drop the use of a guid from the DNS name and instead simple use a prefix (klb) for example. This will mean multiple gateways with the same or different names can now update the record set if their gateway shares a listener with a common host name. 
```

Comment:
Also needs to add the `clusterID` as an annotation/label to any DNSRecords it creates. Should we add that here?

Reply:
Starting to wonder if `ownerID` shouldn't just be added as an optional field to the  DNSRecord.Spec since it, along with "rootHost", are going to be important when it comes to multicluster.

Reply:
might avoid specifics of how it is passed for now